hera_cal/abscal.py

```diff
@@ -2779,126 +2833,276 @@ def get_d2m_time_map(data_times, data_lsts, model_times, model_lsts, unwrap=True
+        ant_offset = np.max(list(data_antpos.keys())) + 1  # increase all antenna indices by this amount
+        joint_antpos = {**data_antpos, **{ant + ant_offset: pos for ant, pos in model_antpos.items()}}
+        joint_reds = redcal.get_reds(joint_antpos, pols=pols, bl_error_tol=tol)
+
+        # filter out baselines not in data or model or between data and model
+        joint_reds = [[bl for bl in red if not ((bl[0] < ant_offset) ^ (bl[1] < ant_offset))] for red in joint_reds]
+        joint_reds = [[bl for bl in red if (bl in data_bl_to_load) or (reverse_bl(bl) in data_bl_to_load)
+                       or ((bl[0] - ant_offset, bl[1] - ant_offset, bl[2]) in model_bl_to_load)
+                       or reverse_bl((bl[0] - ant_offset, bl[1] - ant_offset, bl[2])) in model_bl_to_load] for red in joint_reds]
+        joint_reds = [red for red in joint_reds if len(red) > 0]
+
+        # map baselines in data to unique baselines in model
+        data_to_model_bl_map = {}
+        for red in joint_reds:
+            data_bl_candidates = [bl for bl in red if bl[0] < ant_offset]
+            model_bl_candidates = [(bl[0] - ant_offset, bl[1] - ant_offset, bl[2]) for bl in red if bl[0] >= ant_offset]
+            assert len(model_bl_candidates) <= 1, ('model_is_redundant is True, but the following model baselines are '
+                                                   'redundant and in the model file: {}'.format(model_bl_candidates))
+            if len(model_bl_candidates) == 1:
+                for bl in red:
+                    if bl[0] < ant_offset:
+                        if bl in data_bl_to_load:
+                            data_to_model_bl_map[bl] = model_bl_candidates[0]
+                        elif reverse_bl(bl) in data_bl_to_load:
+                            data_to_model_bl_map[reverse_bl(bl)] = reverse_bl(model_bl_candidates[0])
+                        else:
+                            raise ValueError("Baseline {} looks like a data baseline, but isn't in data_bl_to_load.".format(bl))
+            assert ((len(data_bl_candidates) <= 1)
+                    or (not data_is_redsol)), ('data_is_redsol is True, but the following data baselines are redundant in the ',
+                                               'data file: {}'.format(data_bl_candidates))
+        # only load baselines in map
+        data_bl_to_load = [bl for bl in data_bl_to_load if bl in data_to_model_bl_map.keys()]
+        model_bl_to_load = [bl for bl in model_bl_to_load if (bl in data_to_model_bl_map.values()) 
+                            or (reverse_bl(bl) in data_to_model_bl_map.values())]
+
+    echo("Selected {} data baselines and {} model baselines to load.".format(len(data_bl_to_load), len(model_bl_to_load)), verbose=verbose)
+    return data_bl_to_load, model_bl_to_load, data_to_model_bl_map
+
+
+def build_data_wgts(data_flags, data_nsamples, model_flags, autocorrs, auto_flags, times_by_bl=None,
```

Comment:
I agree w/ AEW here--perhaps best to make `flags` and `nsamples` optional, which get incorporated into `wgts` if fed, but if left `None` then they contribute unity to the wgts.

Reply:
I added a comment to highlight the concerns. 

It's too difficult to make all parameters optional, since the function needs to know about the baselines and the sizes of the arrays. If the user wants to override them with arrays of ones or Falses, they can do that.